File: Python3/프로그래머스/2/178870. 연속된 부분 수열의 합/연속된 부분 수열의 합.py
def solution(sequence, k):
    answer = []
    start = 0
    end = 0;
    now = sequence[start]
    min_length=10000001
    
    while start<=end:
        if now == k:
            if min_length>end-start:
                min_length = end-start
                answer=[start,end]
            now-=sequence[start]
            start+=1
        elif now<k:
            if end+1<=len(sequence)-1:
                end+=1
                now+=sequence[end]
            else:
                break
        else:
            now-=sequence[start]
            start+=1
            
    return answer

File: C++/프로그래머스/2/178870. 연속된 부분 수열의 합/연속된 부분 수열의 합.cpp
#include <string>
#include <vector>

using namespace std;

vector<int> solution(vector<int> sequence, int k) {
    vector<int> answer;
    int start = 0, end=0;
    int min_length=10000001;
    int now = sequence[start];
    while(start<=end){
        if (now == k) {
    if (end - start < min_length) {
        min_length = end - start;
        answer = {start, end};
    }
    now -= sequence[start++];
} else if (now < k) {
    end++;
    if (end < sequence.size()) {
        now += sequence[end];
    } else {
        break;
    }
} else {
    now -= sequence[start++];
}
    }
    return answer;
}